core/state/state_object.go

```diff
@@ -455,6 +455,15 @@ func (s *stateObject) deepCopy(db *StateDB) *stateObject {
 	return obj
 }
 
+// clearStorageCache clears the various storage caches
+// belonging to the state object.
+// It is only used for RPC.
+func (s *stateObject) clearStorageCache() {
```

Comment:
Is this still needed?

Reply:
Good question, @s1na ! 
@s1na , could you answer the guy's question, been a while since he posted it ..

Reply:
Thanks for raising to my attention @s1na .
Note this is only used within `SetStorage` which is a debug function on statedb which clears the state of a contract and sets a new group of slots. If a previous block has cached some slots it can be they are not cleared during this operation.

To give an example: Say in block 1 we set slot 1 and 2. In block 2 we override the complete storage for that contract. Since those values are cached they will be still accessible in block 2.